packages/experimental/Switch/src/Switch.win32.tsx

```diff
@@ -36,26 +37,80 @@ export const Switch = compose<SwitchType>({
     onOffText: Text,
   },
   useRender: (userProps: SwitchProps, useSlots: UseSlots<SwitchType>) => {
+    const [onOffTextTest, setOnOffTextTest] = React.useState(null);
+    const [finalWidth, setFinalWidth] = React.useState<number>(0);
+    const [textBeingTested, setTextBeingTested] = React.useState<textBeingTestedStates>('init');
+    const { onText, offText } = userProps;
+    const toggleContainerRef = React.useRef(null);
     const switchInfo = useSwitch(userProps);
+
     // grab the styled slots
     const Slots = useSlots(userProps, (layer) => switchLookup(layer, switchInfo.state, switchInfo.props));
 
+    /*
+      Controls the rendering of the onText and offText in order to take measurements of what the minWidth should
+      be. Having the correct minWidth prevents the control from "jiggling" when the onText and offText
+      are different sizes.
+    */
+    const measureOnOffTexts = React.useCallback(
+      (measuredWidth) => {
+        switch (textBeingTested) {
+          case 'init':
+            setTextBeingTested('onText');
+            setOnOffTextTest(onText);
+            break;
+          case 'onText':
+            setFinalWidth(0);
+            setTextBeingTested('offText');
+            setOnOffTextTest(offText);
+            break;
+          case 'offText':
+            setFinalWidth(measuredWidth);
+            setTextBeingTested('done');
+            setOnOffTextTest(null);
+            break;
+          case 'done':
+            if (measuredWidth > finalWidth) {
+              setFinalWidth(measuredWidth);
+            }
+            break;
+        }
+      },
+      [textBeingTested, finalWidth, offText, onText],
+    );
+
+    // Each time we reach a new stage in the measureOnOffTexts method we take a measurement.
+    React.useEffect(() => {
+      toggleContainerRef?.current?.measure((_x, _y, width, _height) => {
+        measureOnOffTexts(width);
+      });
+    }, [textBeingTested, measureOnOffTexts]);
+
+    // If the user changes the either the onText or offText props we trigger a re-measurement of the onText and offText.
+    React.useEffect(() => {
+      if (textBeingTested === 'done') {
+        setTextBeingTested('init');
+      }
+    }, [onText, offText]);
+
     // now return the handler for finishing render
     return (final: SwitchProps) => {
       const { label, offText, onText, labelPosition, ...mergedProps } = mergeProps(switchInfo.props, final);
       const onOffText = switchInfo.state.toggled ? onText : offText;
       const displayOnOffText = !!offText || !!onText;
       const isReduceMotionEnabled = AccessibilityInfo.isReduceMotionEnabled;
       const thumbAnimation = isReduceMotionEnabled ? { animationClass: 'Ribbon_SwitchThumb' } : null;
+      const currentOpacity = onOffTextTest ? 0 : 1;
```

Comment:
>  const currentOpacity = onOffTextTest ? 0 : 1; [](http://example.com/codeflow?start=4&length=47)

This would mean that in the weird case that the Switch's on/off text was programmatically changed midway in a session, that the switch would flicker since it would disappear and then come back once it's done measuring

Reply:
Opacity also doesn't hide the component from screen readers so the switch (and I assume the on/off text) would still be readable via a screen reader while it's switching between the text to figure out what width it should have. It's not likely that that would be an issue, but something to note.